Utilities/DataDistribution/Common/include/Common/O2ExtentVector.h

```diff
@@ -0,0 +1,1110 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_CONTAINER_O2EXTENTVECTOR_H_
+#define O2_CONTAINER_O2EXTENTVECTOR_H_
+
+#include <cassert>
+
+#include "template_helpers.h"
+// #include <allocator_helpers.h>
+
+#include <vector>
+#include <algorithm>
+#include <utility>
+#include <cstddef>
+
+#include <iostream>
+
+#pragma once
+
+namespace detail {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// ExtentPosition
+//
+////////////////////////////////////////////////////////////////////////////////////////////////////
+template <class Allocator>
+class Extent {
```

Comment:
could you add reasonable descriptions/explanations to new classes? What are they doing?